android/KMEA/app/src/main/java/com/tavultesoft/kmea/LanguageSettingsActivity.java

```diff
@@ -196,4 +211,32 @@ public void onBackPressed() {
     finish();
   }
 
+  // Fully details the building of this Activity's list view items.
+  static private class FilteredKeyboardsAdapter extends NestedAdapter<com.tavultesoft.kmea.data.Keyboard, Dataset.Keyboards, String> {
+    static final int RESOURCE = R.layout.list_row_layout1;
+
+    public FilteredKeyboardsAdapter(@NonNull Context context, final Dataset storage, final String languageCode) {
+      // Goal:  to not need a custom filter here, instead relying on LanguageDataset's built-in filters.
+      super(context, RESOURCE, storage.keyboards, storage.keyboardFilter, languageCode);
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+      com.tavultesoft.kmea.data.Keyboard kbd = this.getItem(position);
+
+      // If we're being told to reuse an existing view, do that.  It's automatic optimization.
+      if (convertView == null) {
+        convertView = LayoutInflater.from(getContext()).inflate(RESOURCE, parent, false);
+      }
+
+      View view = convertView;
+
+      ImageView img1 = view.findViewById(R.id.image1);
```

Comment:
Apparently `findViewById` can be slow. It is possible to cache this with a `holder` pattern; see example in https://github.com/keymanapp/keyman/blob/master/oem/firstvoices/android/app/src/main/java/com/firstvoices/keyboards/FVKeyboardListAdapter.java#L33

Reply:
👍 

But my line count differential.  😢